src/middleware/insights.ts

```diff
@@ -0,0 +1,36 @@
+import { Middleware } from '.';
+import { InsightsClient } from '../types';
+
+export type InsightsProps = {
+  insightsClient: InsightsClient;
+};
+
+export type CreateInsightsMiddleware = (props: InsightsProps) => Middleware;
+
+export const createInsightsMiddleware: CreateInsightsMiddleware = props => {
+  const { insightsClient } = props;
+
+  return ({ instantSearchInstance }) => {
+    return {
+      onStateChange() {},
+      subscribe() {
+        instantSearchInstance.mainIndex
+          .getHelper()
+          .setQueryParameter('clickAnalytics', true);
+
+        insightsClient(
+          'onUserTokenChange',
+          userToken => {
+            instantSearchInstance.mainIndex
+              .getHelper()
+              .setQueryParameter('userToken', userToken);
+          },
+          { immediate: true }
+        );
+      },
+      unsubscribe() {
+        insightsClient('onUserTokenChange', () => {});
```

Comment:
We should pass `undefined` to prevent the callback from being called [according to the Search Insights implementation](https://github.com/algolia/search-insights.js/blob/ddca3535e657b7fed6adea7eed88eb4f11a5675d/lib/_cookieUtils.ts#L51-L53):

```suggestion
        insightsClient('onUserTokenChange', undefined);
```

Reply:
Nice catch, and it makes me realize there is a bug in search-insights. (passing `undefined` there can make it throw an exception [here](https://github.com/algolia/search-insights.js/blob/ddca3535e657b7fed6adea7eed88eb4f11a5675d/lib/_cookieUtils.ts#L72))